dogs-initial/src/main/java/com/lambdaschool/dogsinitial/DogsinitialApplication.java

```diff
@@ -2,7 +2,9 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
 
+@EnableWebMvc
 @SpringBootApplication
 public class DogsinitialApplication
 {
```

Comment:
Your `NoHandlerFoundException` handling won't work yet because you didn't assign the running Spring application to an `ApplicationContext` and use it for a `DispatcherServlet`. You can do this with the code below:
```
ApplicationContext ctx = SpringApplication.run(DogsinitialApplication.class, args);

DispatcherServlet dispatcherServlet = (DispatcherServlet) ctx.getBean("dispatcherServlet");
dispatcherServlet.setThrowExceptionIfNoHandlerFound(true);
```